src/lib/util/FormatMessage.ts

```diff
@@ -73,7 +85,7 @@ export const formatMessage = async (message: Message) => {
         );
       }
       //Replace the string's code with the formatted code inside a code block
-      return match.replace(theCode, formattedCode);
+      return match.replace(codeContent, '\n' + formattedCode.trim() + '\n');
```

Comment:
This was causing the indentation issue
it used to attempt to replace the codeContent.trim() prior to this, now it changes the entire code block content, to add further context: \`\`\`codeContent\`\`\` that's what codeContent is